### PR TITLE
fix(discord): Fix installation flow from Discord to Sentry

### PR DESCRIPTION
**Pull request by 17hogeju**
Closes https://github.com/getsentry/getsentry/issues/12287

Flow of adding Sentry to Discord from Discord is broken.

### CODE REVIEW

File: src/sentry/integrations/discord/integration.py [hunk 1]
```diff
@@ -308,17 +308,19 @@ def dispatch(self, request, pipeline):
...
+            except Exception as error:
+                logger.info(
+                    "identity.discord.request-token",
+                    extra={
+                        "has_state": has_state,
+                        "guild_id": request.GET["guild_id"],
+                        "code": request.GET["code"],
+                    },
+                )
+                return pipeline.error(error)
```
Comment (scefali) [inline src/sentry/integrations/discord/integration.py:325]: @17hogeju if we get a JSON error we can log it but I think we should let the pipeline continue

File: src/sentry/integrations/discord/integration.py [hunk 2]
```diff
@@ -308,17 +308,19 @@ def dispatch(self, request, pipeline):
...
+        has_state = "state" in request.GET
+        if has_state:
+            try:
+                raw_state = json.loads(request.GET["state"])
+                pipeline.bind_state("use_setup", raw_state.get("useSetup"))
+            except Exception as error:
+                logger.info(
+                    "identity.discord.request-token",
+                    extra={
+                        "has_state": has_state,
```
Comment (scefali) [inline src/sentry/integrations/discord/integration.py:320]: @17hogeju I don't think we need to log `has_state`, it's going to be true

File: src/sentry/integrations/discord/integration.py [hunk 3]
```diff
@@ -308,17 +308,19 @@ def dispatch(self, request, pipeline):
...
+                        "guild_id": request.GET["guild_id"],
+                        "code": request.GET["code"],
```
Comment (scefali) [inline src/sentry/integrations/discord/integration.py:322]: @17hogeju I wouldn't log the code here. `guild_id` is enough

Comment (17hogeju) [reply src/sentry/integrations/discord/integration.py:322]: @scefali  Should I log the error message?

Comment (scefali) [reply src/sentry/integrations/discord/integration.py:322]: @17hogeju yea, logging the error is fine

**Review by scefali — APPROVED** [review]

**Comment by 17hogeju** [comment]
We might be able to follow the flow described in the following PR: https://github.com/getsentry/sentry/pull/57971

File: src/sentry/integrations/discord/integration.py [hunk 4]
```diff
@@ -164,7 +166,13 @@ def build_integration(self, state: Mapping[str, object]) -> Mapping[str, object]
...
             guild_name = guild_id
 
-        discord_user_id = self._get_discord_user_id(str(state.get("code")))
+        discord_config = state.get("discord", {})
+        if isinstance(discord_config, dict):
+            use_configure = discord_config.get("use_configure") == "1"
+        else:
+            use_configure = False
+        url = self.configure_url if use_configure else self.setup_url
+        discord_user_id = self._get_discord_user_id(str(state.get("code")), url)
```
Comment (17hogeju) [inline src/sentry/integrations/discord/integration.py:175]: I don't think this is using the url...

**Review by scefali — APPROVED** [review]

**Pull request merged**